src/test/java/games/strategy/engine/lobby/server/login/RsaAuthenticatorTest.java

```diff
@@ -0,0 +1,71 @@
+package games.strategy.engine.lobby.server.login;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import games.strategy.util.Util;
+
+public class RsaAuthenticatorTest {
+
+  @Test
+  public void testCanProcess() {
+    final Map<String, String> map = new HashMap<>();
+    map.put(RsaAuthenticator.RSA_PUBLIC_KEY, "");
+    assertTrue(RsaAuthenticator.canProcessChallenge(map));
+    map.clear();
+    map.put(RsaAuthenticator.ENCRYPTED_PASSWORD_KEY, "");
+    assertTrue(RsaAuthenticator.canProcessResponse(map));
+    map.clear();
+
+    // Adding a completely unrelated key shouldn't change the outcome
+    map.put(LobbyLoginValidator.HASHED_PASSWORD_KEY, "");
+    assertFalse(RsaAuthenticator.canProcessResponse(map));
+    map.clear();
```

Comment:
Resetting state like this multiple times within a test is a smell that suggests they probably should be separate tests. :smile: